lib/matplotlib/cbook/__init__.py

```diff
@@ -1333,7 +1333,9 @@ def _check_1d(x):
     """Convert scalars to 1D arrays; pass-through arrays as is."""
     # Unpack in case of e.g. Pandas or xarray object
     x = _unpack_to_numpy(x)
-    if not hasattr(x, 'shape') or len(x.shape) < 1:
+    if (not hasattr(x, 'shape') or
+            not hasattr(x, 'ndim') or
```

Comment:
This seems to fix the problem.  and see the test.  However I am not sure if this is really on us, or the `Kernel`, which is missing `ndim`.  Or maybe we should not be relying on ndim.

Reply:
Is `ndim` always `len(shape)`? (I guess not, but otherwise it make make sense to just use that instead?)

Reply:
There are 120 `ndim` in the Python code base (although some are in tests or non-Python-code), so even though we probably should fix it here, in the long run it may be better that `Kernel` gets `ndim`.

Reply:
The problem is that `Kernel` does not really properly behave like an `ndarray`, so it does need to be converted to an array here. We're (ab)using the fact that it does not define `ndim` as a proxy for "I'm not a proper duck type, just happen to have `shape`". Really one would like the `np.asduckarray` that has been discussed in numpy...